@xen-orchestra/web/src/remote-resources/use-xo-task-collection.ts

```diff
@@ -1,30 +1,57 @@
 import { useXoCollectionState } from '@/composables/xo-collection-state/use-xo-collection-state.ts'
 import { convertTaskToCore } from '@/utils/convert-task-to-core.util.ts'
 import { watchCollectionWrapper } from '@/utils/sse.util'
+import { findTaskById } from '@/utils/xo-records/task.util.ts'
+import type { ResourceContext } from '@core/packages/remote-resource/types.ts'
 import { defineRemoteResource } from '@core/packages/remote-resource/define-remote-resource.ts'
 import type { XoTask } from '@vates/types'
-import { computed } from 'vue'
+import { useSorted } from '@vueuse/core'
+import { computed, type Ref } from 'vue'
 
 const ONE_DAY = 24 * 60 * 60 * 1000
 
-const taskFields: (keyof XoTask)[] = ['id', 'start', 'end', 'properties', 'status', 'progress', 'tasks'] as const
+export const taskFields: (keyof XoTask)[] = [
+  'id',
+  'start',
+  'end',
+  'properties',
+  'infos',
+  'warnings',
+  'status',
+  'progress',
+  'tasks',
+  'result',
+] as const
+
+export function createTaskCollectionState<TArgs extends any[] = []>(
+  tasks: Ref<XoTask[]>,
+  context: ResourceContext<TArgs>
+) {
+  const lastDayTasks = computed(() => {
+    const now = Date.now()
+    return tasks.value.filter(task => now - task.start < ONE_DAY).map(task => convertTaskToCore(task))
+  })
+
+  const sortedTasks = useSorted(tasks, (task1, task2) => task2.start - task1.start)
+
+  const getTaskById = (taskId: XoTask['id']) => {
+    return findTaskById(sortedTasks.value, taskId)
+  }
```

Comment:
You can remove this, this getter is already available through `useXoCollectionState`
```suggestion
```